File: Leetcode/dfs_recursion_backtracking/509_Fibonacci/Solution.py
class Solution(object):
    def fib(self, N):
        """
        :type N: int
        :rtype: int
        """
        if N<=0: return 0
        dp = [0]*2
        dp[1] = 1
        for i in xrange(1, N):
            temp = dp[1]
            dp[1] += dp[0]
            dp[0] = temp
        return dp[1]

File: Leetcode/dfs_recursion_backtracking/509_Fibonacci/Solution.java
class Solution {
    public int fib(int N) {
        if(N<=0) return 0;
        int[] dp = new int[2];
        dp[0] = 0; dp[1] = 1;
        for(int i = 1; i<N; ++i){
            int temp = dp[1];
            dp[1] += dp[0];
            dp[0] = temp;
        }
        return dp[1];
    }
}
